### ISSUE TITLE
Search does not expand selection left when appropriate

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#2734]
Consider the following terminal buffer:

```
blah blah tt
```

Searching for "t" will select the second "t" (because of backwards search), doing an incremental search for "tt" will find no results:

![image](https://user-images.githubusercontent.com/2193314/75180610-69c75680-56f1-11ea-831d-5076d799effd.png)

We should cover this edge case where the only result is expanding the selection left. Additionally, if there were 2 results, the selection should be expanded left instead of going to the next one as it's still technically the same match.

**Comment by kumaran-14** [comment]
@Tyriar I'd like to work on this next. What should I look into?

**Comment by Tyriar** [comment]
@kumaran-14 this one will be somewhere in the search addon, right now if we search for `t` and then `tt` in the demo it won't find the above. Before continuing to search for an _incremental_ find previous, we want to check if the selection can just be expanded to the left.

I'd start here:

https://github.com/xtermjs/xterm.js/blob/e2ac4b45d0747ae3d6ab3eb1e3ba9be7cbdec351/addons/xterm-addon-search/src/SearchAddon.ts#L156-L161

**Comment by kena0ki** [comment]
I'm working on this right now.

I think we should switch the value of `startRow` and `startColumn` based on `incremental` like hasNext() does.
https://github.com/xtermjs/xterm.js/blob/e2ac4b45d0747ae3d6ab3eb1e3ba9be7cbdec351/addons/xterm-addon-search/src/SearchAddon.ts#L145-L147
By doing this, the logic stated in the above comment , i.e. L156-L161, results in unnecessary.

**Comment by kena0ki** [comment]
Demo of the above patch.

- Before
![before](https://user-images.githubusercontent.com/29003390/95561558-de1ef200-0a55-11eb-8289-a50ce5ff67d8.gif)
- After
![after](https://user-images.githubusercontent.com/29003390/95561577-e5de9680-0a55-11eb-80c6-9ce8ee53563f.gif)

**Comment by Tyriar** [comment]
@kena0ki looks good 👍

**Comment by kena0ki** [comment]
Sorry,  I found L156-L161 is not unnecessary.
We need to check whether the selection can expand to right or not first. Otherwise, the selection moves around each time words are input.
![buggy](https://user-images.githubusercontent.com/29003390/95834962-98756880-0d78-11eb-962e-70b892757946.gif)

I ended up simply changing search position to selection end after the first search.
The final result is here.
![result](https://user-images.githubusercontent.com/29003390/95836444-67963300-0d7a-11eb-8d6d-aa50b2ed593d.gif)
